### ISSUE TITLE
Publishing missing widgets on Lab

### ISSUE THREAD

**Issue by EfimovVladimir** [twosigma/beakerx#6893]
From classic notebook I can publish and it works fine
https://nbviewer.jupyter.org/gist/anonymous/7fac242abc5f753640e8de106ba40264

On lab publishing I see only code cells and empty outputs
https://nbviewer.jupyter.org/gist/anonymous/e248c8b2b334b4a38cd9ec4beac82d90

**Comment by scottdraves** [comment]
Lab needs to call "save with widget state" before publishing.

**Comment by piorek** [comment]
@scottdraves I've tried to find a way to use action we are using in classical notebook 
 here https://github.com/twosigma/beakerx/blob/master/js/notebook/src/extension/gistPublish/index.ts#L65
but i could not.

Afaik and @mjurowicz-xsolve thinks same this is not supported in Jupyter Lab jet.

We have to implement it ourselves.

**Comment by scottdraves** [comment]
https://github.com/jupyter-widgets/ipywidgets/issues/1005

**Comment by jasongrout** [comment]
> Afaik and @mjurowicz-xsolve thinks same this is not supported in Jupyter Lab jet.

That's correct - it's not yet implemented for the JupyterLab ipywidgets plugin. A pull request would be much appreciated!

**Comment by piorek** [comment]
will have to check that as it is probably fixed as mentioned here https://github.com/jupyter-widgets/ipywidgets/issues/1005#issuecomment-508592284
